### PR TITLE
Refactor + Support revisions for Discovery bulk upload command

### PR DESCRIPTION
**Pull request by seaneganx**
The refactoring and functional changes are included in separate commits and might be easiest to review if viewed separately. Handling of semantic differences in JSON are also separated, and could possibly be removed if the Registry API is updated in the future to handle semantic change detection.

### CODE REVIEW

File: cmd/registry/cmd/upload-bulk-discovery.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/apigee/registry/rpc"
 	rpcpb "github.com/apigee/registry/rpc"
 	discovery "github.com/googleapis/gnostic/discovery"
+	"github.com/nsf/jsondiff"
```
Comment (timburks) [inline cmd/registry/cmd/upload-bulk-discovery.go:27]: Would this also show up in go.mod? Let's include at least that new line and discuss any other differences that seem worth including

**Review by timburks — CHANGES_REQUESTED** [review]
Thanks! Let's add the new dependency to go.mod and go.sum.

File: cmd/registry/cmd/upload-bulk-discovery.go (hunk 1 above)
Comment (seaneganx) [reply cmd/registry/cmd/upload-bulk-discovery.go:27]: Fixed. I just ran `make lite` and let it decide which changes to include.

Comment (timburks) [reply cmd/registry/cmd/upload-bulk-discovery.go:27]: Thanks, I see we picked up a few extra things - we can tighten up expectations about that going forward.

**Review by timburks — APPROVED** [review]

**Pull request merged**